File: a410(two_unknowns_linear_equation)/C++/two_unknowns_linear_equation.cpp
#include <iostream>
#include <iomanip>

int main(int argc, char const *argv[]) {
  std::ios_base::sync_with_stdio(false);
  std::cin.tie(0);
  
  double a, b, c, d, e, f;
  while (std::cin >> a >> b >> c >> d >> e >> f) {
    double delta = a * e - b * d;
    if (delta == 0) {
        if (a * f - c * d == 0) std::cout << "Too many" << '\n';
        else std::cout << "No answer" << '\n';
        continue;
    }
    double delta_x = c * e - b * f;
    double delta_y = a * f - c * d;
    double x = delta_x / delta;
    double y = delta_y / delta;
    std::cout << std::fixed << std::setprecision(2) << "x=" << delta_x / delta << "\ny=" << delta_y / delta << '\n';
  }
  return 0;
}

File: a410(two_unknowns_linear_equation)/Java/TwoUnknownsLinearEquation.java
import java.util.Scanner;


public class TwoUnknownsLinearEquation {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int a = in.nextInt();
        int b = in.nextInt();
        int c = in.nextInt();
        int d = in.nextInt();
        int e = in.nextInt();
        int f = in.nextInt();
        int delta = a * e - b * d;

        if (delta == 0) {
            if (a * f - c * d == 0) {
                System.out.println("Too many");
            } else {
                System.out.println("No answer");
            }
        } else {
            int delta_x = c * e - b * f;
            int delta_y = a * f - c * d;
            double x = (double) delta_x / delta;
            double y = (double) delta_y / delta;
            System.out.printf("x=%.2f%ny=%.2f%n", x, y);
        }

    }
}
